Foundation.hs

```diff
@@ -114,11 +114,10 @@ instance Yesod App where
     makeLogger = return . appLogger
 
     yesodMiddleware =
-        forceSSL (appForceSSL compileTimeAppSettings) . defaultYesodMiddleware
+        whenSSL (sslOnlyMiddleware sessionTimeout) . defaultYesodMiddleware
 
-forceSSL :: Bool -> Handler a -> Handler a
-forceSSL True = sslOnlyMiddleware sessionTimeout
-forceSSL False = id
+whenSSL :: (a -> a) -> (a -> a)
+whenSSL f = if (appForceSSL compileTimeAppSettings) then f else id
```

Comment:
Yup, it's *compileTime*. I.e. it splices the result of the yaml parse into a constant function that you can call from anywhere.